chain-signatures/node/src/protocol/signature.rs

```diff
@@ -665,7 +663,6 @@ impl SignatureManager {
                         ?err,
                         "failed to retry signature generation: trashing presignature"
                     );
-                    failed_presigs.push(presignature);
```

Comment:
if the presignature is not pushed back into storage, then we should not run retry_failed_generation() with the same presignature at all. I think if we don't want to recycle the presignature, then we should pick another unused presignature.

Reply:
It is possible
Check the PR description, I want to rethink how we do retries